specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2020-06-01-preview/rbac.json

```diff
@@ -0,0 +1,610 @@
+      "type": "object",
+      "properties": {
+        "dataActions": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "An array of data actions that are allowed."
+        }
+      },
+      "description": "The set of data plane operations permitted through this Role Definition."
+    },
+    "DefaultErrorResponse": {
+      "properties": {
+        "error": {
+          "properties": {
+            "code": {
+              "description": "Error code.",
+              "type": "string"
+            },
+            "message": {
+              "description": "Error message indicating why the operation failed.",
+              "type": "string"
+            }
+          }
+        }
+      },
+      "description": "An error response from the service."
+    },
+    "SqlRoleDefinitionListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SqlRoleDefinitionGetResults"
+          },
+          "description": "List of Role Definitions and their properties."
+        }
+      },
+      "description": "The relevant Role Definitions."
+    },
+    "SqlRoleAssignmentCreateUpdateParameters": {
+      "type": "object",
+      "properties": {
+        "roleDefinitionId": {
+          "type": "string",
+          "description": "The unique identifier for the associated Role Definition."
+        },
+        "scope": {
+          "type": "string",
+          "description": "The data plane resource path for which access is being granted through this Role Assignment."
+        },
+        "principalId": {
+          "type": "string",
+          "description": "The unique identifier for the associated AAD principal in the AAD graph to which access is being granted through this Role Assignment. Tenant ID for the principal is inferred using the tenant associated with the subscription."
+        }
+      },
+      "x-ms-azure-resource": true,
+      "description": "Role Assignment Properties."
+    },
+    "SqlRoleAssignmentGetResults": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The fully qualified GUID of the Role Assignment containing the resource path and the ID supplied for this Role Assignment during create."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The ID supplied for this Role Assignment during create."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The type of resource."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SqlRoleAssignmentCreateUpdateParameters",
+          "description": "Properties related to the Role Assignment."
+        }
+      },
+      "x-ms-azure-resource": true,
```

Comment:
The resources in our service which are azure-resource usually have attributes like Location, Name, Id, Tags etc. I am not sure this is true for all azure resources or not. Can you check if RoleAssignment supports these, if yes, then let us add ARMResourceProperties (defined in cosmos-db.json) as attributes, which can be done by adding: 
      "allOf": [
        {
          "$ref": "#/definitions/ARMResourceProperties"
        }
      ]

Reply:
I am using ARMProxyResource from cosmos-db.json now